tutorials/integration.qmd

```diff
@@ -0,0 +1,171 @@
+A simple variant can be implemented as follows (assuming uniform distribution of `rand`):
+
+```{julia}
+using LinearAlgebra, Distributions, SpecialFunctions
+function simple_mc_integrate(M::AbstractManifold, f; N::Int = 1000)
+    V = manifold_volume(M)
+    sum = 0.0
+    q = rand(M)
+    for i in 1:N
+        sum += f(M, q)
+        rand!(M, q)
+    end
+    return V * sum/N
+end
+```
+
+``[`manifold_volume`](@ref)``{=commonmark} and ``[`volume_density`](@ref)``{=commonmark} are closely related to each other, though very few sources explore this connection, and some even claiming a certain level of arbitrariness in defining `manifold_volume`.
+Volume is sometimes considered arbitrary because Riemannian metrics on some spaces like the manifold of rotations are defined with arbitrary constants.
+However, once a constant is picked (and it must be picked before any useful computation can be performed), all geometric operations must follow in a consistent way: inner products, exponential and logarithmic maps, volume densities, etc.
+`Manifolds.jl` consistently picks such constants and provides a unified framework, though it sometimes results in picking a different constant than what is the most popular in some sub-communities.
+
+On Lie groups the situation is more complicated. Invariance under left or right group action is a desired property that leads one to consider Haar measures [^Tornier2020].
+It is, however, unclear what are the practical benefits of considering Haar measures over the Lebesgue measure of the underlying manifold, which often turns out to be invariant anyway.
+
+Last point before we turn to consider an example is the matter of integration through charts.
+This is an approach not currently supported by `Manifolds.jl`.
+One has to define a suitable set of disjoint charts covering the entire manifold and use a method for multivariate Euclidean integration.
+Note that ranges of parameters have to be adjusted for each manifold and scaling based on the metric needs to be applied.
+See [^BoyaSudarshanTilma2003] for some considerations on symmetric spaces.
+
+## Distributions
+
+We will now try to verify that volume density correction correctly changes probability density of an exponential-wrapped normal distribution.
+`pdf_tangent_space` represents probability density of a normally distributed random variable $X_T$ in the tangent space $T_p \mathcal{M}$.
+`pdf_manifold` refers to the probability density of the distribution $X_M$ from the tangent space $T_p \mathcal{M}$ wrapped using exponential map on the manifold.
+``[`volume_density`](@ref)``{=commonmark} function calculates the necessary correction.
+
+Note that our simplified `pdf_manifold` assumes that the probability mass of `pdf_tangent_space` outside of (local) injectivity radius is negligible.
+If it was not, a summation over vectors in the tangent plane pointing at `q` would have to be performed.
```

Comment:
This paragraph I do not understand, neither the “pointing at `q`” nor what you mean in general.

Reply:
I've added some details to it, what do you think?